packages/guilded.js/lib/managers/global/RoleManager.ts

```diff
@@ -1,9 +1,132 @@
+import { Collection } from "@discordjs/collection";
 
 /**
  * A class representing a global role manager. At the moment, we don't cache roles in this structure.
  */
-export class GlobalRoleManager extends GlobalManager {
+export class GlobalRoleManager extends CacheableStructManager<number, Role> {
+
+  /**
+   * Create a new role. 
+   * @param serverId The ID of the server.
+   * @param options Role creation options.
+   * @returns Promise that resolves with the newly created role.
+   */
+  create(
+    serverId: string,
+    options: OptionBody<RolesService["roleCreate"]>
+  ): Promise<Role> {
+    return this.client.rest.router.roles
+      .roleCreate({ serverId, requestBody: options })
+      .then((data) => {
+        const newRole = new Role(this.client, data.role);
+        this.cache.set(newRole.id, newRole);
+        return newRole;
+      });
+  }
+
+  /**
+   * Fetch a role by it's ID
+   * @param serverId The ID of the server.
+   * @param roleId ID of the role to fetch.
+   * @param force Whether or not to force a fetch from the API
+   * @returns Promise that resolves with the fetched role.
+   */
+  fetch(serverId: string, roleId: number, force?: boolean): Promise<Role> {
+    if (!force) {
+      const existingRole = this.client.roles.cache.get(roleId);
+      if (existingRole) return Promise.resolve(existingRole);
+    }
+    return this.client.rest.router.roles
+      .roleRead({ serverId, roleId })
+      .then((data) => {
+        const fetchedRole = new Role(this.client, data.role);
+        this.cache.set(fetchedRole.id, fetchedRole);
+        return fetchedRole;
+      });
+  }
+
+  /**
+   * Fetch all roles in a given server
+   * @param serverId The ID of the server.
+   * @returns Promise that resolves with a Collection of fetched roles.
+   */
+  fetchMany(serverId: string): Promise<Collection<number, Role>> {
+    return this.client.rest.router.roles
+      .roleReadMany({ serverId })
+      .then((data) => {
+        const roles = new Collection<number, Role>();
+
+        // Insert all fetched roles into cache
+        for (const role of data.roles) {
+          const fetchedRole = new Role(this.client, role);
+          this.cache.set(fetchedRole.id, fetchedRole);
+          roles.set(fetchedRole.id, fetchedRole);
+        }
+
+        return roles;
+      });
+  }
+
+  /**
+   * Update a role by ID
+   * @param serverId The ID of the server.
+   * @param roleId ID of the role to update.
+   * @param options Role update options.
+   * @returns Promise that resolves with the updated role.
+   */
+  update(
+    serverId: string,
+    roleId: number,
+    options: OptionBody<RolesService["roleUpdate"]>
+  ): Promise<Role> {
+    return this.client.rest.router.roles
+      .roleUpdate({ serverId, roleId, requestBody: options })
+      .then((data) => {
+        const existingRole = this.cache.get(roleId);
+        if (existingRole) return existingRole._update(data.role);
```

Comment:
It seems for `Role#_update`, the return value isn't a Role but instead nothing (this is actually predated from your PR, just hadn't surfaced before). You can solve this with a simple `return this;` at the end of the _update method.

Reply:
This was fixed in the recent commit. The code originally existed there before but it appeared to have not been included in the commit I made. I've added the changes to resolve this issue.